algorithms/sycl/Reduction_ONEAPI.cpp

```diff
@@ -4,42 +4,53 @@
 
 #include <device.h>
 #include <limits>
+#include <iostream>
 
 namespace device {
 
 
-template <typename T> T Algorithms::reduceVector(T *buffer, size_t size, const ReductionType type, void* streamPtr) {
+template <typename T> T Algorithms::reduceVector(T *buffer, size_t size, const ReductionType type, void* queuePtr) {
   if (api == nullptr)
     throw std::invalid_argument("api has not been attached to algorithms sub-system");
 
-  auto rng = internals::computeExecutionRange1D(internals::WARP_SIZE, size);
-  T *red_ptr = (T *)api->allocGlobMem(sizeof(T));
+  auto queue = reinterpret_cast<cl::sycl::queue *>(queuePtr);
```

Comment:
It is because of a type decay. When a SeisSol user requests a default or a new stream from Device I allocate it but return a pointer to void because it is a generic type and any host compiler can handle it. When we come back to the Device, for example, to perform a specific algorithm I cast a void pointer to the actual type. In case of `sycl` it is  a pointer to `queue` type.   For CUDA, it is `cudaStream_t`. For HIP, it is `hipStream_t`. `cudaStream_t` and `hipStream_t` a pointer by themselves.

I know that it is ugly and unsafe. I was thinking to wrap `void*` to a structure and make it as a return type to SeisSol.
```
struct DeviceStream {
  friend class Device;
  void* getRawData() {
    return ptr{};
  }
private:
  void* ptr{};
}
```

`getRawData` will be needed to pass a CUDA stream or SYCL queue to yateto's generated kernels.

I still do not know whether it will make sense. But if do it I will do it in the next PR which is going to be focused on CUDA Graphs